Add execution modes decision docs (#2444)
astronomer/astronomer-cosmos: docs/guides/run_dbt/execution-modes.rst

``airflow_async`` is available for BigQuery as of Cosmos 1.9 and ``watcher`` is available as of Cosmos 1.11.

2. **Execute dbt commands in a container outside of your Airflow environment.** This type of execution mode offers high levels of environment isolation, and also allows you to spin up the Docker or
Kubernetes container in various cloud or on-premises environments

The following diagram shows a decision tree to help you select the right execution mode for your project needs.

.. image:: ../../_static/execution_mode_map.png
   :alt: A diagram illustrating the details about each execution mode in the two categories, "On the Airflow worker or trigger" and "in a container".

On the Airflow worker or triggerer
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

These execution modes offer faster execution times, since your don't need to spin up any extra containers. You can also use Airflow connections via the ``ProfileConfig``. But, these execution modes do not have any, or offer limited, environment isolation. There are four execution mode options that run on the Airflow worker:

> - `local <./airflow-worker/local-execution-mode.html>`_: Default execution mode, but provides no environment isolation. Run ``dbt`` commands using a local ``dbt`` installation (default)

Reviewer: ```suggestion
- `local <./airflow-worker/local-execution-mode.html>`_: Default execution mode in Cosmos 1.x. In this mode, a dbt command is executed for each dbt node (one dbt command per Airflow task). The dbt project is reparsed during every task execution. By default, each task runs a single dbt node and relies on a user-preinstalled dbt. This mode can operate in two ways:
    - No isolation: dbt is installed in the same Python virtual environment as Airflow.
    - Partial isolation: Users create a dedicated Python virtual environment in their Airflow deployment, install dbt there, and configure Cosmos to use it by setting ``ExecutionConfig.dbt_executable_path``.
```

- `virtualenv <./airflow-worker/cosmos-managed-venv.html>`_: Allows you to address package conflicts and an inability to create a venv at build time. Run ``dbt`` commands from Python virtual environments managed by Cosmos. This
- `airflow_async <./airflow-worker/async-execution-mode.html>`_: (Stable since Cosmos 1.9.0) Optimized for worker efficiency if you have long-running dbt commands. Run the dbt resources from your dbt project asynchronously, by submitting the corresponding compiled SQLs to Apache Airflow's `Deferrable operators <https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/deferring.html>`__

In a container
~~~~~~~~~~~~~~

You can also execute dbt commands in a container outside of the Airflow environment. Choosing these kinds of execution modes provides a high degree of isolation, but requires that you can only create Airflow connections with the dbt ``profiles.yml`` file, requires a pre-existing Docker image, and has slower run times, because of container provisioning.

> - `docker <./container/docker.html>`_ : Run ``dbt`` commands from Docker containers managed by Cosmos (requires a pre-existing Docker image)

Reviewer: ```suggestion
- `docker <./container/docker.html>`_ : Run ``dbt`` commands via Docker containers in the Airflow worker node (requires a pre-existing Docker image)
```

- `kubernetes <./container/kubernetes.html>`_: Run ``dbt`` commands from Kubernetes Pods managed by Cosmos (requires a pre-existing Docker image)

> - `watcher_kubernetes <./container/watcher-kubernetes-execution-mode.html>`_: (experimental since Cosmos 1.13.0) Combines the speed of the watcher execution mode with the isolation of Kubernetes. Check the :ref:`watcher-kubernetes-execution-mode` for more details.

Reviewer: Also  requires a pre-existing Docker image
